llvm/include/llvm/Demangle/ItaniumDemangle.h

```diff
@@ -5430,6 +5430,13 @@ Node *AbstractManglingParser<Derived, Alloc>::parseEncoding() {
   if (IsEndOfEncoding())
     return Name;
 
+  // ParseParams may be false in top-level only, when called from parse().
+  if (!ParseParams) {
+    while (consume())
+      ;
```

Comment:
I believe that this will consume any dot suffix too, e.g. the ".123" part of `_Z3foov.123` which is a "vendor specific suffix according to the Itanium spec. I expected c++filt to not skip it when `--no-params` is specified, but apparently it does (and therefore this matches llvm-cxxfilt's behaviour). I'm in two minds as to whether that's a bug or by design. Someone might want to reach out to the GNU mailing list to find out one way or another.

Either way, it probably deserves a test case.